src/tokens/Avatar/Avatar.cairo

```diff
@@ -140,58 +143,21 @@ func isApprovedForAll{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_chec
     return (isApproved,);
 }
 
-@view
-func getXp{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-    tokenId: Uint256
-) -> (xp: Uint256) {
-    alloc_locals;
-    let account = (tokenId.high * 0x100000000000000000000000000000000) + tokenId.low;
-    let (xp) = IExperienceContract.balanceOf(
-        contract_address=EXPERIENCE_CONTRACT, 
-        account=account
-    );
-    return (xp=xp);
-}
-
 @view
 func tokenURI{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
     tokenId: Uint256
 ) -> (tokenURI_len: felt, tokenURI: felt*) {
     alloc_locals;
-    let (xp) = getXp(tokenId);
+    let (exp_addr) = Avatar_experience_contract.read();
+    let (xp) = IExperienceContract.balanceOf(
+        contract_address=exp_addr, 
+        account=tokenId
+    );
     let (progress) = get_progress(xp);
     let (svg) = create_tokenURI(seed=tokenId.low, progress=progress);
     return (tokenURI_len=svg.arr_len, tokenURI=svg.arr);
 }
 
-@view
-func avatarData{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
```

Comment:
@broody we should ideally move this to attributes returned in the tokenURI json, similar to https://github.com/cartridge-gg/non-fungible-football/blob/main/contracts/src/player.cairo#L223

Reply:
Sure, this is used to reconstruct the avatar in frontend. Should really be hidden from attributes I think, will attach it to the metadata tho

Reply:
Ah nm, attributes is good